src/main/java/com/sap/sgs/phosphor/fosstars/data/github/GitHubDataFetcher.java

```diff
@@ -175,34 +170,29 @@ public GitHubDataFetcher(GitHub github, Path base) throws IOException {
   }
 
   /**
-   * Returns the cache with GitHub commits.
-   */
-  public GitHubDataCache<List<GHCommit>> githubCommitsCache() {
-    return githubCommitsCache;
-  }
-
-  /**
-   * Returns a list of commits for a GitHub project using GitHub API.
+   * Returns a number of latest commits.
    *
    * @param project The project.
-   * @param duration A time frame.
-   * @return The list of commits
+   * @param n The number of commits.
+   * @return The list of commits.
    * @throws IOException If something went wrong.
    */
-  public List<GHCommit> githubCommitsFor(GitHubProject project, Duration duration)
-      throws IOException {
-
-    Optional<List<GHCommit>> cachedCommits = githubCommitsCache.get(project);
-    if (cachedCommits.isPresent()) {
-      return cachedCommits.get();
+  public List<GHCommit> githubCommitsFor(GitHubProject project, int n) throws IOException {
+    Objects.requireNonNull(project, "Oh no! The project is null!");
+    if (n <= 0) {
+      throw new IllegalArgumentException("Oh no! The number of commit is not positive!");
     }
 
     try {
-      Date date = Date.from(Instant.now().minus(duration));
       List<GHCommit> commits = new ArrayList<>();
-      for (GHCommit commit : repositoryFor(project).queryCommits().since(date).list()) {
+      for (GHCommit commit : repositoryFor(project).listCommits()) {
```

Comment:
Are these commits sorted? in reverse order?

Reply:
Yes, the comments are sorted by the commit date. Latest commits go first.